src/coreclr/tools/Common/TypeSystem/IL/Stubs/AsyncResumptionStub.cs

```diff
@@ -41,13 +40,62 @@ private MethodSignature InitializeSignature()
 
         public override MethodIL EmitIL()
         {
-            var emitter = new ILEmitter();
-            ILCodeStream codeStream = emitter.NewCodeStream();
+            ILEmitter ilEmitter = new ILEmitter();
+            ILCodeStream ilStream = ilEmitter.NewCodeStream();
 
-            // TODO: match getAsyncResumptionStub from CoreCLR VM
-            codeStream.EmitCallThrowHelper(emitter, Context.GetHelperEntryPoint("ThrowHelpers"u8, "ThrowNotSupportedException"u8));
+            // Ported from jitinterface.cpp CEEJitInfo::getAsyncResumptionStub
+            if (!_owningMethod.Signature.IsStatic)
+            {
+                if (_owningMethod.OwningType.IsValueType)
+                {
+                    ilStream.EmitLdc(0);
+                    ilStream.Emit(ILOpcode.conv_u);
+                }
+                else
+                {
+                    ilStream.Emit(ILOpcode.ldnull);
+                }
+            }
 
-            return emitter.Link(this);
+            foreach (var param in _owningMethod.Signature)
```

Comment:
On the JIT side here, between `this` and other arguments, we also put: 

* generic context arg, if present. 
  It is set to 0, and will not be used by the calee. Continuation has captured the real value and that will be used instead. But we need to match what the calee expects, so we need to push 0 for the context.
* continuation. 
  it is our arg0. Per the ABI of Async* calls it goes right before the other arguments. 
  (unless this is x86)

On x86, as usual, there is a difference and the hidden args go _after_ formal ones and in reverse order. 

--
*Note - the resume stub is an ordinary function, but what we are resuming is an Async method, thus we shuffle the continuation from our arg0 to its predefined/hidden position in the calee signature.

Reply:
Added these arguments